torchdata/datapipes/iter/__init__.pyi

```diff
@@ -0,0 +1,365 @@
+from torchdata.datapipes.iter.util.tararchivereader import TarArchiveReaderIterDataPipe as TarArchiveReader
+from torchdata.datapipes.iter.util.unzipper import UnZipperIterDataPipe as UnZipper
+from torchdata.datapipes.iter.util.xzfilereader import XzFileReaderIterDataPipe as XzFileReader
+from torchdata.datapipes.iter.util.ziparchivereader import ZipArchiveReaderIterDataPipe as ZipArchiveReader
+
+###############################################################################
+# Reference From PyTorch Core
+###############################################################################
+
+__all__ = [
+    "Batcher",
+    "BucketBatcher",
+    "CSVDictParser",
+    "CSVParser",
+    "Collator",
+    "Concater",
+    "Cycler",
+    "DataFrameMaker",
+    "Demultiplexer",
+    "EndOnDiskCacheHolder",
+    "Enumerator",
+    "Extractor",
+    "FSSpecFileLister",
+    "FSSpecFileOpener",
+    "FSSpecSaver",
+    "FileLister",
+    "FileOpener",
+    "Filter",
+    "FlatMapper",
+    "Forker",
+    "GDriveReader",
+    "Grouper",
+    "HashChecker",
+    "Header",
+    "HttpReader",
+    "InMemoryCacheHolder",
+    "IndexAdder",
+    "IoPathFileLister",
+    "IoPathFileOpener",
+    "IoPathSaver",
+    "IterDataPipe",
+    "IterKeyZipper",
+    "IterableWrapper",
+    "JsonParser",
+    "LineReader",
+    "MapKeyZipper",
+    "Mapper",
+    "Multiplexer",
+    "OnDiskCacheHolder",
+    "OnlineReader",
+    "ParagraphAggregator",
+    "ParquetDataFrameLoader",
+    "RarArchiveLoader",
+    "RoutedDecoder",
+    "Rows2Columnar",
+    "SampleMultiplexer",
+    "Sampler",
+    "Saver",
+    "ShardingFilter",
+    "Shuffler",
+    "StreamReader",
+    "TarArchiveReader",
+    "UnBatcher",
+    "UnZipper",
+    "XzFileReader",
+    "ZipArchiveReader",
+    "Zipper",
+]
+
+# Please keep this list sorted
+assert __all__ == sorted(__all__)
+
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, TypeVar, Union
+
+from torch.utils.data import DataChunk, IterableDataset
+from torch.utils.data._typing import _DataPipeMeta
+from torchdata.datapipes.map import MapDataPipe
+
+########################################################################################################################
+# The part below is generated by parsing through the Python files where IterDataPipes are defined.
+# This base template ("__init__.pyi.in") is generated from mypy stubgen with minimal editing for code injection
+# The output file will be "__init__.pyi".
+# Note that, for mypy, .pyi file takes precedent over .py file, such that we must define the interface for other
+# classes/objects here, even though we are not injecting extra code into them at the moment.
+
+from .util.extractor import CompressionType
+
+try:
+    import torcharrow
+except ImportError:
+    torcharrow = None
+
+T_co = TypeVar("T_co", covariant=True)
+
+class IterDataPipe(IterableDataset[T_co], metaclass=_DataPipeMeta):
```

Comment:
This file is the generated .pyi file. There is one issue - the method hints for TorchData functional DataPipes only seem to show up when a TorchData-defined DataPipe is used.

For example, there is only PyTorch Core functional method hint here:
```python
dp = IterableWrapper([1,2,3])  # Only PyTorch Core method hints are shown
dp = dp.map(lambda x: x)  # Only PyTorch Core method hints are shown
dp = dp.filter(lambda _: False)  # Only PyTorch Core method hints are shown
```
There are both PyTorch Core and TorchData functional method hints after `Enumerator` is called:
```python
dp = IterableWrapper([1,2,3])  # Only PyTorch Core method hints are shown
dp = Enumerator(dp)  # From this point on, all method hints are visible
dp = dp.map(lambda x: x)  # All method hints are shown
dp = dp.filter(lambda _: False). # All method hints are shown
```